rfcs/testdriver-features.md

```diff
@@ -0,0 +1,58 @@
+
+This approach is preferable over the [RFC 212](https://github.com/web-platform-tests/rfcs/pull/212), as it is more ergonomic and supports future extensions ergonomically. The list of supported features can be extended when required.
+
+[Prototype](https://github.com/web-platform-tests/wpt/pull/49122).
+
+## Background
+
+[RFC 185](https://github.com/web-platform-tests/rfcs/blob/master/rfcs/testdriver_bidi.md) proposes adding BiDi support to  `testdriver.js`, which involves using [**WebDriverProtocol**](https://github.com/web-platform-tests/wpt/blob/f54cd920024da0857fdc5b036f16a7d1fd8792fd/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L587) instead of [**WebDriverBidiProtocol**](https://github.com/web-platform-tests/wpt/blob/f54cd920024da0857fdc5b036f16a7d1fd8792fd/tools/wptrunner/wptrunner/executors/executorwebdriver.py#L679) for test runners that use the WebDriver protocol. However, this change could cause regressions in tests that don't use the testdriver BiDi API due to unintended side effects of the transport change. This RFC proposes a solution for the issue.
+
+## Details
+
+### Examples
+
+[html test](https://github.com/web-platform-tests/wpt/blob/117958ef4317d8ed16e9ea6ae63da30262f3b875/infrastructure/webdriver/bidi/subscription.html#L6)
+```javascript
+...
+<script src="/resources/testdriver.js?feature=bidi"></script>
+...
+```
+[javascript test](https://github.com/web-platform-tests/wpt/blob/117958ef4317d8ed16e9ea6ae63da30262f3b875/infrastructure/webdriver/bidi/subscription.window.js#L2)
+```javascript
+...
+// META: script=/resources/testdriver.js?feature=bidi
+...
+```
+
+### Required changes
+
+#### Configure testdriver.js based on the `?feature` flag
+
+The `testdriver.js` checks if the `?feature=bidi` query parameter was included using [document.currentScript.src](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) like in the [prototype](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR16). If `bidi` was not enabled, a [descriptive error will be thrown](https://github.com/web-platform-tests/wpt/pull/49122/files#diff-1fe2b624679a3150e5c86f84682c5901b715dad750096a524e8cb23939e5590fR22).
```

Comment:
We need to document in the RFC what the actual syntax of `feature` is. For example is it allowed to supply it multiple times, or is it a (e.g. comma) separated list?

Reply:
I think we need to specify, and implement, the extension to multiple query parameters now. Otherwise if someone does use this for something internal they might pick a different approach and then we'll be stuck.

I don't think you need to answer any of the other questions. Just specify that for forwards compatibility multiple feature parameters are permitted in the query string and matches are based on exact string match.